test/runner/run.ts

```diff
@@ -0,0 +1,162 @@
+      const result = await generateSourceFile(
+        input,
+        {
+          // Enable stage 3 support (ESNext)
+          next: options.next,
+          // Enable React JSX parsing
+          jsx: options.jsx,
+          // Disable web compatibility
+          disableWebCompat: options.disableWebCompat,
+          // Enables implied strict mode
+          impliedStrict: options.impliedStrict
+        },
+        obj.options.module,
+        obj.options.incremental
+      );
+
+      return { obj, result };
+    })
+  );
+
+  console.timeEnd(
+    ColorCodes.GREEN + 'Running ' + ColorCodes.RESET + list.length + ' test cases.' + ColorCodes.yellow + ' Total time'
+  );
+
+  await Promise.all(
+    set.map(async ({ obj, result }: any) => {
+      (obj.newoutput.ast = result), obj.file;
+      obj.newoutput.pretty = '✖ Soon to be open sourced'; //toJs(result);
+    })
+  );
+}
+
+async function generateSourceFile(
+  input: string,
+  options: Options,
+  isModule: boolean,
+  /* TODO */ _isIncremental: boolean
+) {
+  return isModule ? parseModule(input, options) : parseScript(input, options);
+}
+
+export async function generateNewOutput(list: any) {
+  list.forEach((obj: any) => {
+    obj.data = generateOutputBlock(obj.data, obj.newoutput.ast, obj.newoutput.pretty, obj.options);
+  });
+}
+
+export async function writeNewOutput(list: any, update: boolean) {
+  let updated = 0;
+  await Promise.all(
+    list.map((obj: any): any => {
+      const { data, previous, file } = obj;
+      if (data !== previous) {
+        if (update) {
+          ++updated;
+          promiseToWriteFile(file, data);
+        } else {
+          process.exitCode = 1;
+          console.log('Output mismatch for', file);
+          return Promise.resolve();
+        }
+      }
+    })
+  );
+  console.log('Updated', updated, 'files');
+}
+
+function parseTestFile(obj: any): any {
+  const { data, previous } = obj;
+
+  if (!previous) return;
+
+  // find the options
+  const optionsOffset = data.indexOf(Constants.Options);
+  const start1 = data.indexOf(Constants.JsStart, optionsOffset);
+  const end1 = data.indexOf(Constants.JsEnd, optionsOffset);
+
+  // Negative if no opions are set, so we pass a empty obj
+  const options = optionsOffset === -1 ? {} : eval('0||' + data.slice(start1 + Constants.JsStart.length, end1) + '');
+
+  const inputOffset = data.indexOf(Constants.Input);
+  const start = data.indexOf(Constants.JsStart, inputOffset);
+  const end = data.indexOf(Constants.JsEnd, inputOffset);
+  const input = data.slice(start + Constants.JsStart.length, end);
+
+  return { options, input };
+}
+
+function generateOutputBlock(currentOutput: any, ast: any, printed: any, options: any) {
+  ast = JSON.stringify(ast, null, '    ');
+  let outputIndex = currentOutput.indexOf(Constants.Output);
+
+  if (outputIndex < 0) outputIndex = currentOutput.length;
+
+  let diagnosticString = '';
+
+  if (printed !== '✖ Soon to be open sourced') {
```

Comment:
This has changed if you see in the printer branch. and also there exist additional printer code that need to be ported over here for the printer tests to run.

Reply:
merge that branch to main?

Reply:
I think you already did ;)

Reply:
ahhh...I guess no. I just rebased the latest main - it was auto-merged since no new commits.

your latest code is in the wip-printer repo, and not pushed to the repo?

Reply:
eh yes, and do not merge anything from the wip-printer repo. Only the printer test code.